Spigot-Server-Patches/0696-Add-TameableDeathMessageEvent.patch

```diff
@@ -0,0 +1,33 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Owen1212055 <23108066+Owen1212055@users.noreply.github.com>
+Date: Fri, 19 Mar 2021 23:25:52 -0400
+Subject: [PATCH] Add TameableDeathMessageEvent
+
+
+diff --git a/src/main/java/net/minecraft/world/entity/EntityTameableAnimal.java b/src/main/java/net/minecraft/world/entity/EntityTameableAnimal.java
+index 74636c4c0851da1cd5732daa0a2d1a4edda2050f..cf0caedd24aa63b2872bdd8ecc6a5b1f0d3e322f 100644
+--- a/src/main/java/net/minecraft/world/entity/EntityTameableAnimal.java
++++ b/src/main/java/net/minecraft/world/entity/EntityTameableAnimal.java
+@@ -3,6 +3,8 @@ package net.minecraft.world.entity;
+ import java.util.Optional;
+ import java.util.UUID;
+ import javax.annotation.Nullable;
++import io.papermc.paper.adventure.PaperAdventure;
++import io.papermc.paper.event.entity.TameableDeathMessageEvent;
+ import net.minecraft.SystemUtils;
+ import net.minecraft.advancements.CriterionTriggers;
+ import net.minecraft.nbt.NBTTagCompound;
+@@ -185,7 +187,12 @@ public abstract class EntityTameableAnimal extends EntityAnimal {
+     @Override
+     public void die(DamageSource damagesource) {
+         if (!this.world.isClientSide && this.world.getGameRules().getBoolean(GameRules.SHOW_DEATH_MESSAGES) && this.getOwner() instanceof EntityPlayer) {
+-            this.getOwner().sendMessage(this.getCombatTracker().getDeathMessage(), SystemUtils.b);
++            // Paper start - TameableDeathMessageEvent
++            TameableDeathMessageEvent event = new TameableDeathMessageEvent((org.bukkit.entity.Tameable) getBukkitEntity(), PaperAdventure.asAdventure(this.getCombatTracker().getDeathMessage()));
++            if (event.callEvent()) {
++                this.getOwner().sendMessage(PaperAdventure.asVanilla(event.getDeathMessage()), SystemUtils.b);
```

Comment:
Maintainers might ask you to use an obfhelper for `SystemUtils.b`.

Reply:
There is a OBFHELPER already. SystemUtils#getNullUUID() so just use it please.

```suggestion
+                this.getOwner().sendMessage(PaperAdventure.asVanilla(event.getDeathMessage()), SystemUtils.getNullUUID());
```